src/cljs/athens/events.cljs

```diff
@@ -175,6 +177,17 @@
     (update db :athena/open not)))
 
 
+(reg-event-fx
+  :athena/create-page
+  (fn [_ [_ title uid open-on-sidebar]]
+    {:fx [[:dispatch [:page/create title uid]]
+          [:dispatch [:athena/toggle]]
+          (if
+            open-on-sidebar
+            [:dispatch-later {:ms 300 :dispatch [:right-sidebar/open-item uid]}]
```

Comment:
Not a huge fan of artificial delays, what will happen when DB is big enough for it not to complete in 300ms?

Reply:
I was unable to solve this though. How can I wait until `:page/create` finishes? I can look more into it though

Reply:
The solution here is to rework `:right-sidebar/open-item` to only update the `db`, and create a subscription that does the `datascript` pulling and element sorting. 

That way the subscription will be updated once the `:page/create` transactions is done. Thoughts?

Reply:
I could do it here, or in another PR